assets/src/edit-story/utils/textMeasurements.js

```diff
@@ -65,18 +67,28 @@ export function calculateFitTextFontSize(element, width, height) {
   // Binomial search for the best font size.
   let minFontSize = 1;
   let maxFontSize = PAGE_HEIGHT;
+  let margin;
   while (maxFontSize - minFontSize > 1) {
-    const mid = (minFontSize + maxFontSize) / 2;
-    setStyles(measurer, { fontSize: `${mid}px` });
-    const currentHeight = measurer.scrollHeight;
-    if (currentHeight > height) {
+    const mid = dataPixels((minFontSize + maxFontSize) / 2);
+    const { marginOffset } = calcFontMetrics({ ...element, fontSize: mid });
```

Comment:
Would be great to add some comments (here and elsewhere) for the future to note why we need this, that we're cutting off the part added by the line-height, etc.